test/Microsoft.TestPlatform.AcceptanceTests/TranslationLayerTests/DiscoverTests.cs

```diff
@@ -196,31 +196,37 @@ public void DiscoverTestsUsingSourceNavigation(RunnerInfo runnerInfo)
     }
 
     [TestMethod]
-    // flaky on the desktop runner, desktop framework combo
-    [NetFullTargetFrameworkDataSource(useDesktopRunner: false)]
+    [NetFullTargetFrameworkDataSource(inProcess: true)]
     [NetCoreTargetFrameworkDataSource]
     public void CancelTestDiscovery(RunnerInfo runnerInfo)
     {
         // Setup
         var testAssemblies = new List<string>
         {
-            GetAssetFullPath("DiscoveryTestProject.dll"),
             GetAssetFullPath("SimpleTestProject.dll"),
-            GetAssetFullPath("SimpleTestProject2.dll")
+            GetAssetFullPath("SimpleTestProject2.dll"),
         };
 
         SetTestEnvironment(_testEnvironment, runnerInfo);
         Setup();
 
         var discoveredTests = new List<TestCase>();
         var discoveryEvents = new Mock<ITestDiscoveryEventsHandler>();
-        discoveryEvents.Setup((events) => events.HandleDiscoveredTests(It.IsAny<IEnumerable<TestCase>>())).Callback
-            ((IEnumerable<TestCase> testcases) => discoveredTests.AddRange(testcases));
+        discoveryEvents.Setup(events => events.HandleDiscoveredTests(It.IsAny<IEnumerable<TestCase>>()))
+            .Callback((IEnumerable<TestCase> testcases) => discoveredTests.AddRange(testcases));
+        discoveryEvents.Setup(events => events.HandleDiscoveryComplete(It.IsAny<long>(), It.IsAny<IEnumerable<TestCase>>(), It.IsAny<bool>()))
```

Comment:
Doesn't change much for the discovery cancelled case but without this setup the tested condition was always true even when not cancelling.

Reply:
Comment is no longer relevant as we now check that we correctly signal abortion.